include/vicon/udp.h

```diff
@@ -228,28 +242,52 @@ class ObjectReference
 template<typename ObjectDataType = ObjectData>
 class UDPClient
 {
-public:
-    UDPClient() {}
+private:
+    /*
+     * NB: The first member of this struct *should* be object ID, but the
+     * Vicon system seems to just give a value of zero for every item, which
+     * isn't terribly helpful. So instead we distinguish objects based on their
+     * names.
+     */
+#ifdef _MSVC
+#pragma pack(push, 1)
+    struct RawObjectData {
+        uint8_t unused;
+        uint16_t itemDataSize;
+        char objectName[24];
+        double position[3], attitude[3];
+    };
+    #pragma pack(pop)
+#else
+    struct __attribute__((__packed__)) RawObjectData {
+        uint8_t unused;
+        uint16_t itemDataSize;
+        char objectName[24];
+        std::array<double, 3> position, attitude;
```

Comment:
I think this would be cleaner if you use an 'old fashioned' array for both - not sure what the bug you're working around here is but I don't think it's worth it :smile:

Reply:
That's what I meant to do... the ``#ifdef`` is just because it's a packed struct and you do this differently in gcc and MSVC

Reply:
Ahh ok - I was wondering if it was some sort of combination of a VC++ bug regarding packing std::array/std::array having some extra debug members on VC++ (which I **think** could be allowed). https://gist.github.com/PhilCK/1534763 could be nicer

Reply:
Good idea.